_posts/2020-01-13-serverless-express-api.md

```diff
@@ -0,0 +1,562 @@
+---
+layout: post
+title: Build & Deploy a Serverless Express API to AWS
+date: 2020-01-13
+tags: javascript tutorial serverless sam
+author: Matthew Tyler
+---
+
+# Introduction
+
+Released in 2015, AWS API Gateway allows developers to build and publish APIs that can be consumed by clients over public internet and virtual private cloud networks. At Re:Invent 2019, announced a significant addition to the service, that they have called "HTTP APIs". The HTTP API is a direct response to customers who wanted to build simple HTTP backed API's, who did not need the complete feature set of API Gateway. The new HTTP API service is perfect for hosting lamba backend applications that are written in popular HTTP Frameworks like Express, Flask, .NET etc.
```

Comment:
I think there's an important "just because you can, doesn't mean you should' statement required here.

Reading this at the moment, it reads that this a a 'suggested' way of building applications.  I agree that lowering the bar to entry is a good thing - but we should also highlight the technical debt that this approach will create.

Who's this post for?
Where should they go to learn more/refactor once they have more experience?
What issues are they likely to run into?

Reply:
I feel like there's a long response to that question, and results in asking further questions that are not immediately relevant to what I was trying to get across in this post, which was showing how to use express with the new HTTP API feature. There's additional posts coming to show how to use the JWT authorizer functionality, and refactoring to publish an OpenAPI document, but that is too much for one post.

I don't think there is anything inherent in the tech choice here that guarantees technical debt. Yes, it could occur, but that will be because of bad practice in the code base, via some mix of coupling between models and poor domain boundaries. That's a post unto itself - and a better one to make later.

This is aimed at rank beginners, and was somewhat formulated by looking at tags on dev.to. It's deliberately written to drive traffic.

Reply:
Agree that digging deeper is a whole separate post.  But I think we should be signposting it - this isn't how *we* write apps.

Reply:
Found a good spot to put it in the first paragraph.

The difficult part is saying it in way that isn't too pretentious - remembering this is for newbs.

Reply:
:+1: 

When we write the posts on how we do it, then we can update the reference.